src/js/libgeo/GeoBasics.js

```diff
@@ -154,13 +161,27 @@ function addElementNoProof(el) {
 
     // Detect unsupported operations or missing or incorrect arguments
     var op = geoOps[el.type];
+    var isSet = false;
+    var getKind = function(el) {
+        return csgeo.csnames[el].kind;
```

Comment:
I think it would be better to name that argument `name`, since `el` is typically the object for a geometric element, not its name. Since you are only using this in a single location, I personally would prefer an anonymous function at the point of use, so one doesn't have to scroll to relate these two.